crates/payload/builder/src/service.rs

```diff
@@ -237,13 +235,14 @@ where
     /// This also takes a stream of chain events that will be forwarded to the generator to apply
     /// additional logic when new state is committed. See also
     /// [`PayloadJobGenerator::on_new_state`].
-    pub fn new(generator: Gen, chain_events: St) -> (Self, PayloadBuilderHandle<T>) {
+    pub fn new(generator: Gen, chain_events: St, max_payload_jobs: usize) -> (Self, PayloadBuilderHandle<T>) {
```

Comment:
I'd like to keep this fn as is and instead move this to a builder style setter fn